### ISSUE TITLE
Pattern `postTypes` no longer honored

### ISSUE THREAD

**Issue by webmandesign** [WordPress/gutenberg#52717]
### Description

In my block theme I set `'postTypes' => [ 'wp_template', 'wp_template_part' ]` (via [`register_block_pattern()` PHP function](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-patterns/#register_block_pattern)) for all patterns that are meant to be used in Site Editor only (they should not be used in page block editor).

Then I insert such pattern into a Template Part with `<!-- wp:pattern {"slug":"my-theme/site/header"} /-->`, for example.

Everything is working perfectly fine with WordPress 6.2.2 - the site header pattern is correctly rendered when editing site header template part in Site Editor.

However, when I activate Gutenberg 16.2.1, the **pattern is not rendered at all in Site Editor when editing related template part**. The pattern is not available via block/pattern inserter either.

I've traced the issue down to setting up the `postTypes` attribute. Once I remove this attribute from the pattern registration arguments, the pattern renders OK.

Please check and fix this functionality. I think it should still work the same as it does with WP6.2.

### Step-by-step reproduction instructions

The description says it all.

### Environment info

- WordPress 6.2.2
- Gutenberg 16.2.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by andrewserong** [comment]
Thanks for reporting this issue @webmandesign! I tested and confirmed that the issue was present in Gutenberg 16.2.1, and has been resolved in `trunk`. It appears to have been fixed by #52503, which sets the template type for the site editor by retrieving the current post type via `getEditedPostType`. It seems that prior to that PR, the site editor's settings had a template type of `undefined`, so the match against `postTypes` for registered block patterns was failing.

Since this fix will be rolled out to both Gutenberg 16.3 and WP 6.3, I'll close out this issue now.
